tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_jvm_compile.py

```diff
@@ -42,10 +58,43 @@ def test_if_runtime_classpath_exists(self):
     runtime_classpath.add_for_targets([target], [('default', pre_init_runtime_entry)])
 
     task = self.create_task(context)
-    resulting_classpath = task.create_runtime_classpath()
+    resulting_classpath = task.create_classpath_product()
     self.assertEqual([('default', pre_init_runtime_entry), ('default', compile_entry)],
       resulting_classpath.get_for_target(target))
 
+  def test_modulized_targets_not_compiled_for_export_classpath(self):
+    init_subsystems([JvmPlatform])
+    targets = self.make_linear_graph(['a', 'b', 'c', 'd', 'e'], target_type=JavaLibrary)
+    context = self.context(target_roots=[targets['a'], targets['c']], options={'jvm-platform': {'compiler': 'dummy'}})
+    context.products.get_data('compile_classpath', ClasspathProducts.init_func(self.pants_workdir))
+    # This should cause the jvm compile execution to exclude target roots and their
+    # dependess from the set of relevant targets.
+    context.products.require_data("export_dep_as_jar_classpath")
+    self.execute(context)
+    export_classpath = context.products.get_data('export_dep_as_jar_classpath')
+    # assert none of the modulized targets have classpaths.
+    self.assertFalse(export_classpath.get_for_target(targets['a']) + export_classpath.get_for_target(targets['b']) + export_classpath.get_for_target(targets['c']))
+    self.assertTrue(export_classpath.get_for_target(targets['d']))
+    self.assertTrue(export_classpath.get_for_target(targets['e']))
```

Comment:
assertTrue/False could be a bit confusing, looks like `get_for_target` should return list of tuples: https://github.com/pantsbuild/pants/blob/9dea18f41dd5d1608375ddf0bc78742c04abcf35/src/python/pants/backend/jvm/tasks/classpath_products.py#L223-L232

so `assertTrue(len(export_classpath.get_for_target(..) == 0)` would be more clear. 

In case a None is expected in some other cases, there is `assertIs/IsNotNone`

Reply:
Asserting truthy-ness on empty iterables in python is pretty standard I think, but happy to make it more explicit.

Reply:
To be clear that function returns an empty list when it doesn't have any classpath products for a target.

Reply:
Gotcha. Sounds good